### PR TITLE
Vue naming style

### PR DESCRIPTION
**Pull request by mcnelson**
First effort at establishing some kind of naming pattern for our Vue apps.

Want to get this out there sooner than later before it grows too much for us to be motivated.

Still WIP for now.

### CODE REVIEW

File: style/vue/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+- Component filenames are ProperCased and mixins are snake_cased.
+- Components live entirely in `src/components/` and `src/views/`.
+- Components are resourcefully named when possible: index, show, new, edit.
+- Entry-point components live at the `src/views/{resource}/` level and not any deeper.
+- Child components live at the `src/views/{resource}/` level or deeper.
+- Prefix child components and partially reused components in `src/components` with an underscore `_`.
+
+### Generic Components
+
+**Globally reusable** components, relevant to any area of your app, live in `src/components`. Components that are generic to **one area** of the app and cannot be easily reused anywhere else also live in `src/components`, but are prefixed with an underscore as in `_Component.vue`.
```
Comment (feitla) [inline style/vue/README.md]: these components we are not underscoring

File: style/vue/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    path: 'foo-bars',
+    name: 'FooBars',
+    component: FooBarIndex
+  },
+]
+```
+
+The component `FooBarIndex` would live in `src/views/FooBars/Index.vue`.
+
+### Child components
```
Comment (feitla) [inline style/vue/README.md:37]:
```suggestion
### Child components/Partials
```

**Comment by mcnelson** [comment]
Let's get Grant's :+1: on this, Luda if you want to review again, and we'll get this puppy merged.

File: style/vue/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+Vue Naming
+==========
+
+Vue is unopinionated about how to name files within the project. This can lead to calamity as different developers introduce varying conventions. This proposes a **resourceful** approach to component file naming.
+
+### At a glance
+
+- Component filenames are ProperCased and mixins are snake_cased.
```
Comment (feitla) [inline style/vue/README.md]: not snake_cased but camelCased

File: style/vue/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+The component `FooBarIndex` would live in `src/views/FooBars/Index.vue`.
+
+### Child components
+
+Components that are imported by any entry point component, sometimes referred to as "partials," live in `src/views/{resourceful name}/`. Use further directories to group children when it makes sense.
+
+Child component filenames start with a `_`, such as `_FormActions.vue`. This convention helps clarify a parent-child relationship between two or more components.
+
+Mixins also live one level deep in a resourceful directory, for example `src/views/{resourceful name}/formatters.js`. Mixin filenames are snake cased.
```
Comment (feitla) [inline style/vue/README.md]: mixin filenames are cameCased

**Review by feitla — CHANGES_REQUESTED** [review]

**Review by feitla — APPROVED** [review]

**Pull request merged**